application/common/service_type.py

```diff
@@ -0,0 +1,69 @@
+from aws_lambda_powertools import Logger
+from common.change_event_exceptions import ValidationException
+from common.constants import (
+    DENTIST_SERVICE_KEY,
+    ORGANISATION_SUB_TYPES_KEY,
+    ORGANISATION_TYPES_KEY,
+    PHARMACY_SERVICE_KEY,
+    SERVICE_TYPES,
+    VALID_SERVICE_TYPES_KEY,
+)
+
+logger = Logger(child=True)
+
+
+class ServiceType:
+    """
+    Class for the service type.
+    """
+
+    name: str
+    organisation_type_id: str
+    organisation_sub_type: list[str]
+    valid_service_types: set[int]
+
+    def __init__(self, organisation_type_id: str):
+        self.name = self.find_service_type(organisation_type_id)
+        self.organisation_type_id = SERVICE_TYPES[self.name][ORGANISATION_TYPES_KEY]
+        self.organisation_sub_type = SERVICE_TYPES[self.name][ORGANISATION_SUB_TYPES_KEY]
+        self.valid_service_types = SERVICE_TYPES[self.name][VALID_SERVICE_TYPES_KEY]
+        logger.debug(f"ServiceType: {self.name}")
+
+    def __repr__(self) -> str:
+        return f"<ServiceType: name={self.name} valid_service_types={self.valid_service_types}>"
+
+    def find_service_type(self, organisation_type_id: str) -> str:
+        """Select the service type based on the organisation type id
+
+        Args:
+            organisation_type_id (str): organisation type id of NHS entity
+
+        Raises:
+            ValidationException: if the organisation type id is not found in the valid service types
+
+        Returns:
+            str: service type name
+        """
+        if SERVICE_TYPES[PHARMACY_SERVICE_KEY][ORGANISATION_TYPES_KEY] == organisation_type_id:
```

Comment:
to avoid having to re-write this function when we add new types can we take the service types data and spin in round to a lookup from org to service type key

```
PHARMACY_SERVICE_KEY = "PHARMACY"
DENTIST_SERVICE_KEY = "DENTIST"
ORGANISATION_TYPES_KEY = "ORGANISATION_TYPES"
ORGANISATION_SUB_TYPES_KEY = "ORGANISATION_SUB_TYPES"
VALID_SERVICE_TYPES_KEY = "VALID_SERVICE_TYPES"
SERVICE_TYPES = {
    PHARMACY_SERVICE_KEY: {
        ORGANISATION_TYPES_KEY: "PHA",
        ORGANISATION_SUB_TYPES_KEY: ["COMMUNITY"],
        VALID_SERVICE_TYPES_KEY: {13, 131, 132, 134, 137},
    },
    DENTIST_SERVICE_KEY: {
        ORGANISATION_TYPES_KEY: "Dentist",
        ORGANISATION_SUB_TYPES_KEY: ["TBA"],
        VALID_SERVICE_TYPES_KEY: {12},
    },
}



lookup = {}
for key in SERVICE_TYPES:

    
    lookup[SERVICE_TYPES[key][ORGANISATION_TYPES_KEY]] = key
    

def getServiceTypeKey(orgType:str) -> str:
    if orgType in lookup:
       return lookup[orgType]
    else:
      raise Exception(
                f"Unexpected Org Type ID '{orgType}'"
            )

print(getServiceTypeKey("PHA"))
```